Retrieve license information from distribution

dlax opened the pull request:
Closes #42

Following a recent [thread](https://lists.okfn.org/pipermail/ckan-dev/2017-February/010794.html) on the mailing list.
We have been using this patch for a couple of times; I finally got to finish the patch and submit.

metaodi commented on ckanext/dcat/profiles.py:
```diff
@@ -769,6 +777,19 @@ def parse_dataset(self, dataset_dict, dataset_ref):
+                            self._object_value(license, DCT.title))
+                    if license_id is not None:
+                        dataset_dict['license_id'] = license_id
+
```
I would propose to put this whole block in a method called `_license()` to make it very easy to override this behavior.

dlax replied:
Done in de220a425c2fb08edf6dcd2b4db3bf7ea5ba5336

amercader commented on ckanext/dcat/profiles.py:
```diff
@@ -307,6 +308,35 @@ def _spatial(self, subject, predicate):
+        '''
+        license_uri2id = {}
+        license_title2id = {}
+        for license_id, license in LicenseRegister().items():
```
This only needs to be run once right? There's no need to run the same loop on every dataset (and there could be potentially loads of them when harvesting). So you could do something like:

```python 
if not self._license_uri2id:
    for license_id, license in LicenseRegister().items():
         self._license_uri2id[license.url] = license_id
         # ...
```

dlax commented:
Adrià Mercader a écrit :
> Thanks @dlax <https://github.com/dlax>, see just one minor comment. Also
> this should definitely be documented in the README. I suggest adding a
> note on the mappings table
> <https://github.com/ckan/ckanext-dcat#rdf-dcat-to-ckan-dataset-mapping>
> ("See note about dataset license") and expand on what's the default
> behaviour and that can people can customize it overriding the |_license|
> method on their own profile.

Added two commits addressing these points.

amercader commented:
Thanks @dlax, great stuff
